src/xlsb.rs

```diff
@@ -188,36 +188,48 @@ impl<RS: Read + Seek> Xlsb<RS> {
         };
         let mut buf = vec![0; 1024];
         let mut number_formats = BTreeMap::new();
+        let mut xfs = Vec::new();
 
-        // Search for BrtBeginFmts
-        if iter.next_skip_blocks(0x0267, &[], &mut buf).is_ok() {
-            let len = read_usize(&buf);
-
-            for _ in 0..len {
-                let _ = iter.next_skip_blocks(0x002C, &[], &mut buf)?; // BrtFmt
-                let fmt_code = read_u16(&buf);
-                let fmt_string = wide_str(&buf[2..], &mut 0)?.into_owned();
-                number_formats.insert(fmt_code, fmt_string);
-            }
-
-            // Search for BrtBeginCellXFs
-            if iter.next_skip_blocks(0x0269, &[], &mut buf).is_ok() {
-                let len = read_usize(&buf);
+        loop {
+            match iter.read_type()? {
+                0x0267 => {
+                    // BrtBeginFmts
+                    let _len = iter.fill_buffer(&mut buf)?;
+                    let len = read_usize(&buf);
 
-                for _ in 0..len {
-                    let _ = iter.next_skip_blocks(0x002F, &[], &mut buf)?; // BrtXF
-                    let num_fmt = read_u16(&buf[2..4]);
-                    self.formats.push(match number_formats.get(&num_fmt) {
-                        Some(fmt) if is_custom_date_format(fmt) => CellFormat::Date,
-                        None if is_builtin_date_format_id(num_fmt.to_string().as_bytes()) => {
-                            CellFormat::Date
-                        }
-                        _ => CellFormat::Other,
-                    });
+                    for _ in 0..len {
+                        let _ = iter.next_skip_blocks(0x002C, &[], &mut buf)?; // BrtFmt
+                        let fmt_code = read_u16(&buf);
+                        let fmt_string = wide_str(&buf[2..], &mut 0)?.into_owned();
+                        number_formats.insert(fmt_code, fmt_string);
+                    }
+                }
+                0x0269 => {
+                    // BrtBeginCellXFs
+                    let _len = iter.fill_buffer(&mut buf)?;
+                    let len = read_usize(&buf);
+                    for _ in 0..len {
+                        let _ = iter.next_skip_blocks(0x002F, &[], &mut buf)?; // BrtXF
+                        xfs.push(read_u16(&buf[2..4]));
+                    }
                 }
+                // BrtEndStyleSheet
+                0x0117 => break,
```

Comment:
we could probably early exit when reaching [BrtEndCellXFs](https://learn.microsoft.com/en-us/openspecs/office_file_formats/ms-xlsb/cccbca18-dcb2-4235-a123-827cf1ab11d2) ?

Reply:
I think, we can exit after loop in BrtStartCellXFs, as you suggest below.